Src/WitsmlExplorer.Api/HttpHandlers/AuthorizeHandler.cs

```diff
@@ -1,58 +1,40 @@
 
 namespace WitsmlExplorer.Api.HttpHandlers
 {
     public static class AuthorizeHandler
     {
-        public static async Task<IResult> Authorize([FromServices] ICredentialsService credentialsService, HttpRequest httpRequest)
+        public static async Task<IResult> Authorize([FromQuery(Name = "keep")] bool keep, [FromServices] ICredentialsService credentialsService, HttpContext httpContext)
         {
-            string basicAuth = httpRequest.Headers[WitsmlClientProvider.WitsmlTargetServerHeader];
-            return Results.Ok(await credentialsService.ProtectBasicAuthorization(basicAuth));
-        }
-
-        public static async Task<IResult> AuthorizeAndSetCookie([FromServices] ICredentialsService credentialsService, IHttpContextAccessor httpContextAccessor)
-        {
-            string basicAuth = httpContextAccessor?.HttpContext?.Request.Headers[WitsmlClientProvider.WitsmlTargetServerHeader];
+            string basicAuth = httpContext?.Request.Headers[EssentialHeaders.WitsmlTargetServer].ToString();
             ServerCredentials creds = await credentialsService.GetCredentialsFromHeaderValue(basicAuth);
-            string encryptedPassword = await credentialsService.ProtectBasicAuthorization(basicAuth);
 
-            CookieOptions cookieOptions = new()
-            {
-                SameSite = SameSiteMode.Strict,
-                MaxAge = TimeSpan.FromDays(1),
-                Secure = true,
-                HttpOnly = true
-            };
-            string cookieValue = Convert.ToBase64String(Encoding.UTF8.GetBytes($"{creds.UserId}:{encryptedPassword}"));
-            httpContextAccessor.HttpContext.Response.Cookies.Append(Uri.EscapeDataString(creds.Host.ToString()), cookieValue, cookieOptions);
-            return Results.Ok(encryptedPassword);
-        }
-
-        public static IResult AuthorizeWithCookie([FromServices] ICredentialsService credentialsService, IHttpContextAccessor httpContextAccessor)
-        {
-            ServerCredentials targetServer = httpContextAccessor?.HttpContext?.Request.GetWitsmlServerHttpHeader(WitsmlClientProvider.WitsmlTargetServerHeader, n => "");
-            IRequestCookieCollection cookies = httpContextAccessor.HttpContext.Request.Cookies;
-            if (targetServer.Host == null || !cookies.TryGetValue(Uri.EscapeDataString(targetServer.Host.ToString()), out string cookie))
+            if (creds.IsCredsNullOrEmpty())
             {
                 return Results.Unauthorized();
             }
-            if (!credentialsService.ValidEncryptedBasicCredentials($"{cookie}@{targetServer.Host}"))
+            else
             {
-                return Results.Unauthorized();
+                string encryptedCredentials = await credentialsService.ProtectBasicAuthorization(basicAuth);
+                CookieOptions cookieOptions = new()
+                {
+                    SameSite = SameSiteMode.Strict,
+                    MaxAge = keep ? TimeSpan.FromDays(1) : null,
```

Comment:
"Session cookies are deleted when the current session ends. The browser defines when the "current session" ends, and some browsers use session restoring when restarting. **This can cause session cookies to last indefinitely**." https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies
This means that by not checking "keep me logged in", we actually set a cookie that last indefinitely in some cases. Maybe we need to rethink how we should use cookies with that in mind.

Reply:
Ok, let's address that :)